Add support for additional rules and other QoL updates

mgmeyers opened the pull request:
Hey @Clemens-E, thanks for the great plugin!

I apologies for the large PR, but there were a few things that were making this plugin difficult to use, and I thought I'd knock them out all at once.

Here's what I did:

1. Added settings for toggling rule categories, and enabling /  disabling specific rules (re: #6)
2. Added support for "picky mode"
2. Added additional styling to match the language tool app. Specifically, I added a "style" underline
3. Updated the underlines to more closely match language tool (hope this is ok, let me know if you'd like me to revert these changes)
4. Fixed some issues around the widget not opening when clicking on marks, and not disappearing when clicking elsewhere
5. Enhanced support for a local languagetool server so that this plugin works offline when a local server is running
6. Added a command to clear marks

<img width="1064" alt="Screen Shot 2021-05-16 at 11 26 21 AM" src="https://user-images.githubusercontent.com/2694747/118408240-8d015d00-b639-11eb-98e1-adc85f8c46cf.png">

Clemens-E commented:
crazy pr! 
I guess you did most things on my to-do list.
Looks all very good and probably better than would have done it 😄 

> Enhanced support for a local languagetool server so that this plugin works offline when a local server is running

Just don't get this, what change exactly? I'm running my own LanguageTool Server and also ran one locally, both work fine.

I will review/merge this tomorrow, thank you for the pr!
CI is failing but its just some stupid yarn hash issue that I had across all my repositories, don't know why.

mgmeyers commented:
I deleted the obsidian entry from the `yarn.lock` file and reran `yarn install`, and it seems to have done the trick in terms of getting the build to pass!

> Just don't get this, what change exactly? I'm running my own LanguageTool Server and also ran one locally, both work fine.

Specifically this line: https://github.com/Clemens-E/obsidian-languagetool-plugin/blob/master/src/SettingsTab.ts#L27

This was failing for me when I was offline. Updated it to use the settings URL: https://github.com/Clemens-E/obsidian-languagetool-plugin/pull/11/files#diff-de4a7265f5c03dfe889e27e7e63eda0ef9bbeb271ffc966c6385c31a927fbc4cL27-R69

Clemens-E commented:
I did the same thing on the master branch (delete the lock file and install again)

Ah, well, I tried the local server before adding that "select default language" thing, so that wasn't on my radar.
